xeofs/preprocessing/sanitizer.py

```
@@ -85,17 +86,30 @@ def transform(self, X: DataArray) -> DataArray:
         # Check if input has the correct coordinates
         self._check_input_coords(X)
 
-        # Store sample coordinates for inverse transform
-        self.sample_coords_transform = X.coords[self.sample_name]
-        # Remove NaN entries; only consider full-dimensional NaNs
-        # We already know valid features from the fitted dataset
-        X = X.isel({self.feature_name: self.is_valid_feature})
-        # However, we need to recheck for valid samples, as the new dataset may
-        # have different samples
-        is_valid_sample = ~X.isnull().all(self.feature_name).compute()
-        X = X.isel({self.sample_name: is_valid_sample})
-        # Store valid sample locations for inverse transform
-        self.is_valid_sample_transform = is_valid_sample
+        X_valid_features = self._get_valid_features(X)
+
+        # Remove full-dimensional NaN entries
+        X = X.dropna(dim=self.sample_name, how="all")
+        X = X.dropna(dim=self.feature_name, how="all")
+
+        # For new data only, validate that NaN features match the original
+        if self.has_run:
+            if not X_valid_features.equals(self.is_valid_feature):
+                raise ValueError(
+                    "Input data had NaN features in different locations than"
+                    " than the original data."
+                )
+
+        # Only carry out isolated NaN check for non-dask-backed data
+        if not data_is_dask(X):
```

May as well let the SVD fail on its own when its time to compute if we're able to build everything else lazily